hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/collection/impl/ReactiveBasicCollectionPersister.java

```diff
@@ -0,0 +1,179 @@
+ */
+package org.hibernate.reactive.persister.collection.impl;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.hibernate.HibernateException;
+import org.hibernate.MappingException;
+import org.hibernate.cache.CacheException;
+import org.hibernate.cache.spi.access.CollectionDataAccess;
+import org.hibernate.collection.spi.PersistentCollection;
+import org.hibernate.engine.spi.SharedSessionContractImplementor;
+import org.hibernate.internal.CoreMessageLogger;
+import org.hibernate.mapping.Collection;
+import org.hibernate.persister.collection.BasicCollectionPersister;
+import org.hibernate.persister.spi.PersisterCreationContext;
+import org.hibernate.pretty.MessageHelper;
+import org.hibernate.reactive.pool.ReactiveConnection;
+import org.hibernate.reactive.session.ReactiveConnectionSupplier;
+import org.hibernate.reactive.util.impl.CompletionStages;
+
+import org.jboss.logging.Logger;
+
+public class ReactiveBasicCollectionPersister extends BasicCollectionPersister implements ReactiveCollectionPersister {
+	private static final CoreMessageLogger LOG = Logger.getMessageLogger(
+			CoreMessageLogger.class,
+			ReactiveBasicCollectionPersister.class.getName()
+	);
+	private static final String VALUE_STR = "(?, ?)";
+
+	public ReactiveBasicCollectionPersister(
+			Collection collectionBinding,
+			CollectionDataAccess cacheAccessStrategy,
+			PersisterCreationContext creationContext) throws MappingException, CacheException {
+		super( collectionBinding, cacheAccessStrategy, creationContext );
+	}
+
+	private ReactiveConnection getReactiveConnection(SharedSessionContractImplementor session) {
+		return ( (ReactiveConnectionSupplier) session ).getReactiveConnection();
+	}
+
+	public CompletionStage<Void> recreateReactive(
+			PersistentCollection collection,
+			Serializable id,
+			SharedSessionContractImplementor session)
+			throws HibernateException {
+		ReactiveConnection reactiveConnection = getReactiveConnection( session );
+
+		if ( isInverse ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( !isRowInsertEnabled() ) {
+			return CompletionStages.voidFuture();
+		}
+
+		if ( LOG.isDebugEnabled() ) {
+			LOG.debugf(
+					"Inserting collection: %s",
+					MessageHelper.collectionInfoString( this, collection, id, session )
+			);
+		}
+
+		// create all the new entries
+		Iterator entries = collection.entries( this );
+
+		if ( entries.hasNext() ) {
+			collection.preInsert( this );
+
+			List<Object> valuePairList = new ArrayList<>();
+
+			while ( entries.hasNext() ) {
+				valuePairList.add( id );
+				valuePairList.add( entries.next() );
+			}
+
+			// In order to insert element collection values create a native query with placeholders followed by
+			// value sets
+			return reactiveConnection
+					.update(
+							getSQLInsertElementCollectionString(
+									valuePairList.size() / 2 ),
```

Comment:
Why do you need `valuePairList.size() / 2 ` ? I don't understand the reason.
Does ORM do the same?

Reply:
I followed the basic logic within AbstractCollectionPersister and converted to using the reactiveConnection.update(..) call.

https://github.com/hibernate/hibernate-orm/blob/15e59b3690debe7f08b41b2ab73eb42b273a42dc/hibernate-core/src/main/java/org/hibernate/persister/collection/AbstractCollectionPersister.java#L1286
is an example for recreate()...   Also used remove() and insertRows() logic to re-implement.